resources/todomvc/architecture-examples/svelte/src/Item.svelte

```diff
@@ -0,0 +1,51 @@
+<script>
+    import { createEventDispatcher, tick } from 'svelte';
+
+    const dispatch = createEventDispatcher();
+
+    function removeItem() {
+        dispatch('removeItem', { index });
+    }
+
+    function startEdit() {
+        dispatch('startEdit', { index });
+    }
+
+    function handleEdit(event) {
+        dispatch('handleEdit', { key: event.key, target: event.target})
+    }
+
+    function updateItem(event) {
+        dispatch('updateItem', { text: event.target.value, index });
+    }
+
+    function toggleItem(event) {
+        dispatch('toggleItem', { index, checked: event.target.checked });
+    }
+
+    async function focusInput(element) {
+		await tick();
+		element.focus();
+	}
+
+    export let item;
+    export let editing;
+    export let index;
+</script>
+
+<li class="{item.completed ? 'completed' : ''} {editing === index ? 'editing' : ''}">
+    <div class="view">
+        <input class="toggle" type="checkbox" on:change={toggleItem} checked={item.completed} />
+        <!-- svelte-ignore a11y-label-has-associated-control -->
+        <label on:dblclick={startEdit}>{item.description}</label>
+        <button on:click={removeItem} class="destroy" />
+    </div>
+
+    {#if editing === index}
+        <div class="input-container">
+            <!-- svelte-ignore a11y-autofocus -->
+            <input value={item.description} id="edit-todo-input" class="edit" on:keydown={handleEdit} on:blur={updateItem} use:focusInput />
```

Comment:
maybe we can use `bind:value` on `item.description` and avoid all the churn around the `blur` event. The drawback is that the item would be changed at each keystroke, so I'm not sure this would work. Also I'm not sure how to handle the `Escape` key.

Reply:
The parent component evaluates the key and decides what to do with the escape key.

Reply:
I mean, I'm not sure how to handle the Escape key with the change I'm proposing :-)

Reply:
Probably just leave it as-is then? I assume the "correct" todomvc behavior is to only change on blur / change and not on keypress